sdk/python/kfp/components/tasks_group.py

```diff
@@ -85,7 +85,7 @@ def _make_name_unique(self):
 
         group_id = pipeline_context.Pipeline.get_default_pipeline(
         ).get_next_group_id()
-        self.name = f'{self.group_type}-{group_id}'
+        self.name = f'{self.group_type.value}-{group_id}'
```

Comment:
Nice find! Looks like the issue may have been related to [this](https://www.andy-pearce.com/blog/posts/2023/Jan/whats-new-in-python-311-new-and-improved-modules/#strenum-and-reprenum).